k4Reco/GaudiLumiCalClusterer/include/LumiCalClusterer.h

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2020-2024 Key4hep-Project.
+ *
+ * This file is part of Key4hep.
+ * See https://key4hep.github.io/key4hep-doc/ for further info.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef LumiCalClusterer_h
+#define LumiCalClusterer_h 1
+
+// clustering options
+#define _CLUSTER_MIDDLE_RANGE_ENGY_HITS 1
+#define _MOLIERE_RADIUS_CORRECTIONS 1
+#define _CLUSTER_MIXING_ENERGY_CORRECTIONS 1
+
+// verbosity
+#define _GENERAL_CLUSTERER_DEBUG 0
+#define _CREATE_CLUSTERS_DEBUG 0
+#define _CLUSTER_BUILD_DEBUG 0
+#define _VIRTUALCLUSTER_BUILD_DEBUG 0
+#define _MOL_RAD_CORRECT_DEBUG 0
+
+#include "Global.h"
+
+#include "GlobalMethodsClass.h"
+#include "LCCluster.h"
+
+#include <DDSegmentation/BitFieldCoder.h>
+
+#include <Gaudi/Algorithm.h>
+
+#include <edm4hep/CalorimeterHitCollection.h>
+#include <edm4hep/SimCalorimeterHitCollection.h>
+
+#include <memory>
+#include <string>
+
+enum RETVAL {
+  NOK = 0,
+  OK = 1,
+};
+
+class LumiCalClustererClass {
+
+public:
+  // Constructor
+  LumiCalClustererClass(const Gaudi::Algorithm* alg);
+  LumiCalClustererClass(LumiCalClustererClass const& rhs) = delete;
+  LumiCalClustererClass& operator=(LumiCalClustererClass const& rhs) = delete;
+  LumiCalClustererClass(LumiCalClustererClass&& rhs) = delete;
+  LumiCalClustererClass& operator=(LumiCalClustererClass&& rhs) = default;
+  ~LumiCalClustererClass() = default;
+
+  // initialization routine - Called at the begining of the job.
+  void init(GlobalMethodsClass const& gmc);
+
+  /// set the cutOnFiducialVolume flag
+  void setCutOnFiducialVolume(bool cutFlag) { m_cutOnFiducialVolume = cutFlag; }
+
+  void createDecoder(const std::string& decoderString);
+
+  // main actions in each event -Called for every event - the working horse.
+  std::pair<RETVAL, edm4hep::CalorimeterHitCollection> processEvent(edm4hep::SimCalorimeterHitCollection& col);
```

Comment:
Should this really be taking a `SimCalorimeterHitCollection` instead of a `CalorimeterHitCollection`? What is the reason to not pass this by `const&` here?

Reply:
I think so, it's basically the function that runs with the input of `operator()` of the algorithm, in this case it just happens not to be implemented in the main algorithm... I added `const`.